packages/cli/src/modules/migrate/commands/versions/migrate.ts

```diff
@@ -94,20 +96,32 @@ export async function migrateMovedPackages(options?: {
         }
 
         const movedPackageName = packageInfo.backstage?.moved;
+        if (!movedPackageName) {
+          continue;
+        }
 
-        if (movedPackageName) {
-          movedPackages.set(depName, movedPackageName);
-          console.log(
-            chalk.yellow(
-              `Found a moved package ${depName}@${depVersion} -> ${movedPackageName} in ${pkgName} (${depType})`,
-            ),
-          );
+        movedPackages.set(depName, movedPackageName);
+        console.log(
+          chalk.yellow(
+            `Found a moved package ${depName}@${depVersion} -> ${movedPackageName} in ${pkgName} (${depType})`,
+          ),
+        );
+
+        didPackageChange = true;
+        didAnythingChange = true;
 
-          didPackageChange = true;
-          didAnythingChange = true;
+        delete depsObj[depName];
 
-          depsObj[movedPackageName] = depsObj[depName];
-          delete depsObj[depName];
+        if (depName.startsWith('@backstage/') && depVersion === 'backstage:^') {
+          const info = await fetchPackageInfo(movedPackageName);
+          depsObj[movedPackageName] = `^${info['dist-tags'].latest}`;
+        } else if (
+          movedPackageName.startsWith('@backstage/') &&
+          hasYarnPlugin
+        ) {
+          depsObj[movedPackageName] = 'backstage:^';
+        } else {
```

Comment:
Feels like the order of these two conditions should be swapped? 🤔

Reply:
I think `movedPackageName.startsWith('@backstage/') && hasYarnPlugin` is the least likely option, while `depName.startsWith('@backstage/') && depVersion === 'backstage:^'` is the most likely option of these cases -- besides the else-case.

The check for the package name might be unnecessary, true.

```ts
if (depVersion === 'backstage:^') {
} else if (hasYarnPlugin && movedPackageName.startsWith('@backstage/')) {
} else {
}
```